### PR TITLE
Methods for transforming to and from field-aligned coordinates

### PR DESCRIPTION
**Pull request by johnomotani**
Includes changes in #104, which ideally should be merged before reviewing this.

Thanks to @friva000 from whose routines I lifted the fft interpolation logic.

### CODE REVIEW

**Comment by johnomotani** [comment]
There are lots of PEP8 complaints from `xbout/tests/test_boutdataarray.py` about too-long lines, but I think it's best to ignore them. Those lines are effectively a table of expected results, which are much easier to read without splitting the lines.

**Comment by ZedThree** [comment]
Might be a way to comment the region to say "ignore this bit"?

**Comment by johnomotani** [comment]
> Might be a way to comment the region to say "ignore this bit"?

It's possible, but as far as I can see (https://flake8.pycqa.org/en/latest/user/violations.html) requires a comment on each line. That's not very nice, but I guess is better in the long run as it means we can run flake8 over the whole project and not get errors that we want to ignore. I'll add the suppressions.

**Comment by johnomotani** [comment]
:+1: suppressing the errors that we want to ignore showed a too-long line error that needed fixing but which I had missed in the noise before, so I think it is a good idea. Thanks @ZedThree!

File: xbout/tests/test_boutdataset.py [hunk 1]
```diff
@@ -55,6 +56,35 @@ def test_test_method(self, tmpdir_factory, bout_xyt_example_files):
...
+        n = ds['n']
+        n.attrs['direction_y'] = 'Standard'
+        n_aligned_from_array = n.bout.toFieldAligned()
+
+        # check n_aligned does not exist yet
+        try:
+            ds['n_aligned']
+            assert False
+        except KeyError:
+            pass
```
Comment (ZedThree) [inline xbout/tests/test_boutdataset.py]: Is this just `assert 'n_aligned' not in ds`?

**Comment by ZedThree** [comment]
I feel like there's probably a more compact way to write the tests, but these are nice and explicit at least!

File: xbout/tests/test_boutdataset.py (hunk 1 above)
Comment (johnomotani) [reply xbout/tests/test_boutdataset.py]: Yes it is. Sorry I pulled the trigger a bit too quick on this merge. I'll stick the fix in the next PR. Thanks @ZedThree!

**Pull request merged**